cpp/include/cudf_test/column_wrapper.hpp

```diff
@@ -94,7 +94,7 @@ struct fixed_width_type_converter {
   template <typename FromT                                                        = From,
             typename ToT                                                          = To,
             typename std::enable_if<std::is_same<FromT, ToT>::value, void>::type* = nullptr>
-  ToT operator()(FromT element) const
+  __host__ __device__ ToT operator()(FromT element) const
```

Comment:
Very strange issue. 
We have `CUDA_HOST_DEVICE_CALLABLE` that can be used here. Another option I've seen used is to mark the function as `constexpr`, which is (I think) preferred when it can be used. Can you please try this option?

Reply:
I'm confused why this is even happening.  These things are only being called on the host, unless I'm reading this wrong.

Reply:
`constexpr` works.

Reply:
> I'm confused why this is even happening. These things are only being called on the host, unless I'm reading this wrong.

I don't get it either. At least constexpr is not out of place in this context.

Reply:
Something else is going on here.  If the constexpr fixes it, I think we're just masking another issue.

Reply:
I have no idea too...

Reply:
To me it looks like this functor is only used to initialize `thrust::host_vector`. I have dug around `thrust` trying to confirm whether `host_vector` is initialized on host or device, with no success.

Reply:
It's definitely host.  I'm trying it locally.   However, if I add __device__ it does compile - it just kills the process the first time it calls the functos.